src/glossary.md

```diff
@@ -1,5 +1,29 @@
 # Glossary
 
+## Terms used in the Specification
+
+### Ill-formed Program
+
+A Rust Program is ill-formed if it violates a statically set constraint placed on it by the language.
+Unless otherwise statement explicitly an implementation shall issue an appropriate diagnostic for an ill-formed program and shall not translate such a program.
+
+#### No Diagnostic Required
+
+When No Diagnostic is Required for a static constraint, the implementation may issue a diagnostic for the violation, and may translate the program.
+If such a program is succesfully translated, no constraints are placed on the result of executing the program.
+
+> [!NOTE]
+> The constraint which is modified by the phrase "No Diagnostic Required" may be defined implicitly, such as with the term "shall", rather than explicitly stating that the program is ill-formed.
```

Comment:
For now, since our tooling isn't ready, we can't use the admonitions here. If we want to avoid blocking merges, I would recommend just using the existing form, like this:

```suggestion
> **Note**:  The constraint which is modified by the phrase "No Diagnostic Required" may be defined implicitly, such as with the term "shall", rather than explicitly stating that the program is ill-formed.
```

Reply:
This isn't really being used except for the proper spec chapters, which also rely on the tooling, so I'm ok with this being blocked on the tooling support as well.